ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,103 @@ const base = url.substring(0, url.lastIndexOf('/'));
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(true);
+      const nuxeoError = layout.$$('nuxeo-error');
+      expect(nuxeoError.code).to.equal('404');
+      expect(nuxeoError.url).to.equal('notfound.html');
+      expect(nuxeoError.message).to.equal('Failed to find layout');
+    });
+
+    test('Should hide layout when href is empty', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href=""></nuxeo-layout>
+        `,
+      );
+      expect(isElementVisible(layout)).to.be.false;
+    });
   });
 
-  test('layout invalid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Stamping', () => {
+    test('Should stamp the layout when model is defined', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
+      `);
+      await flush();
+
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      const span = layout.element.$$('span');
+      expect(isElementVisible(layout.$$('nuxeo-error'))).to.be.false;
+      expect(layout.element.localName).to.equal('dummy-layout');
+      expect(span.innerText).to.exist.and.to.equal('dummy');
+    });
+
+    test('Should stamp the layout when model is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model=""></nuxeo-layout>
+      `);
+      await flush();
+
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      expect(layout.element.$$('span')).to.exist;
+      expect(layout.element.$$('nuxeo-input')).to.exist;
+    });
   });
 
-  test('layout invalid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Validation', () => {
+    test('Should invalidate the layout when required input is empty', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid"}'></nuxeo-layout>
+      `);
+      const validity = await layout.validate();
+      expect(validity).to.be.false;
+    });
+
+    test('Should validate the layout when fields are valid', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid", "data": "foo"}'></nuxeo-layout>
+      `);
+      const validity = await layout.validate();
```

Comment:
We should add a spy on the validate method of the dummy-layout `element` to make sure that it is called this time, since this is part of our contract.